### PR TITLE
Implements addAdbPublicKey API endpoint

### PR DESCRIPTION
**Pull request by neofreko**
Reason: manually approve adb connection from web UI does not scale for test automation. Especially those that are using different adb keys on each session (eg: via container)

This changes also added swagger doc endpoint. Using `stf local` command, it will end up at: `http://localhost:7106/docs/`

### CODE REVIEW

File: lib/units/api/controllers/user.js [hunk 1]
```diff
@@ -400,3 +401,56 @@ function getUserAccessTokens(req, res) {
...
+    .then(function(key) {
+      return dbapi.lookupUsersByAdbKey(key.fingerprint)
+        .then(function(cursor) {
+          return cursor.toArray()
+        })
+        .then(function(users) {
+          return {
+            key: {
+              title: data.title || key.comment
+              , fingerprint: key.fingerprint
```
Comment (sorccu) [inline lib/units/api/controllers/user.js]: Small thing but could you reduce the indentation on every line that starts with a comma? The fields should line up.

**Comment by denis99999** [comment]
Is this feature will be merged soon ?

File: lib/units/api/controllers/user.js [hunk 2]
```diff
@@ -400,3 +401,56 @@ function getUserAccessTokens(req, res) {
...
+      ])
+    })
+    .catch(dbapi.DuplicateSecondaryIndexError, function() {
+      // No-op
+      return res.json({
+        success: true
+      })
+    }).catch(function(err) {
+      log.error('Failed to insert new adb key fingerprint: ', err.stack)
+      return res.status(500).json({
```
Comment (koral--) [inline lib/units/api/controllers/user.js]: Shouldn't we distinguish cases when supplied key is malformed and respond with error 400 in such case?

**Comment by denis99999** [comment]
Actually, ADB keys are created through the UI, so for consistency is this feature includes the updating of the UI while creating ADB keys through the API ?

**Comment by stevelibuzz** [comment]
It is a great API enhancement, do you consider adding a new API to delete the ADBKey or purge all adbKeys?

File: lib/units/api/controllers/user.js (hunk 2 above)
Comment (neofreko) [reply lib/units/api/controllers/user.js]: adbkit does not provide a specific exception object we can catch. This is what adbkit do when encountering such input:
```
reject new Error "Unrecognizable public key format"
```

What we can do better here probably is passing through the error message. Would that be suffice?

**Comment by neofreko** [comment]
> Actually, ADB keys are created through the UI, so for consistency is this feature includes the updating of the UI while creating ADB keys through the API ?

@denis99999 
This PR will also trigger the ADB keys update event. Current ADB keys management on UI part is depending on the same mechanism, AFAIK. So it'll be in sync.

**Comment by neofreko** [comment]
> It is a great API enhancement, do you consider adding a new API to delete the ADBKey or purge all adbKeys?

@stevelibuzz 
Good point there, ADB keys may accumulate over time. Unfortunately, I don't have that kind of use case at the moment. Feel free to create follow up PR ;)

File: lib/units/api/controllers/user.js (hunk 2 above)
Comment (denis99999) [reply lib/units/api/controllers/user.js]: @koral-- , is it possible now to merge soon this interesting feature ?

Comment (koral--) [reply lib/units/api/controllers/user.js]: Sorry for the delay.
LGTM

> What we can do better here probably is passing through the error message. Would that be suffice?

Stacktrace from that error is printed to the log so I think it is OK to leave it as is.

**Comment by sorccu** [comment]
Finally merged! Thanks :)

**Pull request merged**